eq-author/src/components/QuestionnaireTable/Row.js

```diff
@@ -268,76 +377,14 @@ export const Row = ({
         onClick={handleClick}
         data-test="table-row"
       >
-        <TD>
-          <QuestionnaireLink
-            data-test="anchor-questionnaire-title"
-            title={displayName}
-            onClick={handleLinkClick}
-            to={buildQuestionnairePath({
-              questionnaireId: id,
-            })}
-          >
-            {shortTitle && (
-              <ShortTitle>
-                <Truncated>{shortTitle}</Truncated>
-              </ShortTitle>
-            )}
-            <Truncated>{title}</Truncated>
-          </QuestionnaireLink>
-        </TD>
-        <TD>{createdBy.displayName}</TD>
-        <TD>
-          <FormattedDate date={createdAt} />
-        </TD>
-        <TD>
-          <FormattedDate date={updatedAt} />
-        </TD>
-        <TD>
-          <Permissions>
-            <Permission>View</Permission>
-            <Permission disabled={!hasWritePermission}>Edit</Permission>
-          </Permissions>
-        </TD>
-        <TD>
-          <Tooltip content={locked ? "Locked" : "Not locked"} place="top">
-            <IconTextButton
-              title="Lock"
-              onClick={handleLock}
-              data-test="lockButton"
-              disabled={!hasWritePermission}
-            >
-              {locked ? <LockedIcon /> : <UnlockedIcon />}
-            </IconTextButton>
-          </Tooltip>
-        </TD>
-        <TD>
-          <Tooltip content={starred ? "Starred" : "Not starred"} place="top">
-            <IconTextButton
-              title="Star"
-              onClick={handleStar}
-              data-test="starButton"
-            >
-              {starred ? <StarredIcon /> : <UnstarredIcon />}
-            </IconTextButton>
-          </Tooltip>
-        </TD>
-        <TD>
-          <div onFocus={handleButtonFocus} data-test="action-btn-group">
-            <ButtonGroup>
-              <DuplicateQuestionnaireButton
-                data-test="btn-duplicate-questionnaire"
-                onClick={handleDuplicateQuestionnaire}
-                hideText
-              />
-              <IconButtonDelete
-                hideText
-                data-test="btn-delete-questionnaire"
-                onClick={handleDeleteQuestionnaire}
-                disabled={!canDelete}
-              />
-            </ButtonGroup>
-          </div>
-        </TD>
+        {renderEnabled(tableHeadings, Headings.TITLE)}
+        {renderEnabled(tableHeadings, Headings.OWNER)}
+        {renderEnabled(tableHeadings, Headings.CREATED)}
+        {renderEnabled(tableHeadings, Headings.MODIFIED)}
+        {renderEnabled(tableHeadings, Headings.PERMISSIONS)}
+        {renderEnabled(tableHeadings, Headings.LOCKED)}
+        {renderEnabled(tableHeadings, Headings.STARRED)}
+        {renderEnabled(tableHeadings, Headings.ACTIONS)}
```

Comment:
If you did filter out stuff (see following comment), then this would simplify to a simple `map`. Could probably replace it with iteration even if you didn't, though (it's doing the same-ish thing 8 times)